default/AI/ShipDesignAI.py

```diff
@@ -334,7 +335,12 @@ def _check_cache_for_consistency(self):
             print_error(e)
 
         corrupted = []
-        for designname in self.design_id_by_name:
+        # create a copy of the dict-keys so we can alter the dict
+        for designname in list(self.design_id_by_name.keys()):
```

Comment:
Seems to me the list() is redundant; in Python 2 dict.keys() already makes a list